airmozilla/manage/tests/views/test_picturegallery.py

```diff
@@ -242,6 +242,27 @@ def test_picture_add(self):
         assert picture.size
         eq_(picture.modified_user, self.user)
 
+        event = Event.objects.get(title='Test event')
+        response = self.client.get(url, {'event': event.id})
+        eq_(response.status_code, 200)
+        response = self.client.get(url, {'event': 9999})
+        eq_(response.status_code, 404)
+
+        url = reverse('manage:picturegallery')
+        response = self.client.get(url)
+        eq_(response.status_code, 200)
+
+        # check delete works properly
+        event.placeholder_img = event.placeholder_img
+        event.picture = picture
+        event.save()
+        eq_(Picture.objects.all().count(), 1)
+        delete_url = reverse('manage:picture_delete', args=(picture.id,))
+        response = self.client.post(delete_url, data={'event': event.id})
+        eq_(response.status_code, 200)
+        eq_(Picture.objects.filter(event=event).count(), 0)
+        eq_(Picture.objects.all().count(), 0)
+
```

Comment:
This all should be a separate test. There are basically two ways you can add a picture to the gallery. 
1) Just adding a picture. Plain and simple. 
2) Having an event in mind when you're adding a picture. 

For example, click on the event edit of an event somewhere in the event manager. For the picture there should be a link there that says "Pick another" or something. It takes you to the picture gallery management page **but** with the `?event=12345` thing added to the URL. And when you arrive on the gallery page it says "You are here to associate a picture with TITLE OF EVENT".

So you're there to pick a picture. If you want to add a new picture (instead of picking one) we need to remember which event ID it was you were focusing on. That's what this is about. 

So, I would start a new test called something like `test_picture_add_with_event`. 
You do the same `event = Event.objects.get(title='Test event')` and `response = self.client.get(url, {'event': event.id})` but this time you check that the title is in the `response.content` on the form for uploading a picture. 

And you should check that there's a link there on that page back to the picture gallery WITH that `?event=12345` in it.